### PR TITLE
SERVO_ENDSTOPS is a -tuple, not an enabling flag

### PR DESCRIPTION
**Pull request by Wackerbarth**
Addresses Issue#2537

### CODE REVIEW

File: Marlin/Conditionals.h [hunk 1]
```diff
@@ -525,6 +525,7 @@
       #define Z_ENDSTOP_SERVO_NR -1
     #endif
     #if (X_ENDSTOP_SERVO_NR >= 0) || (Y_ENDSTOP_SERVO_NR >= 0) || (Z_ENDSTOP_SERVO_NR >= 0)
+      #define HAS_SERVO_ENDSTOPS true
```
Comment (thinkyhead) [inline Marlin/Conditionals.h]: This can just be defined, doesn't need to be set to `true`. But… (see below)…

File: Marlin/Marlin_main.cpp [hunk 2]
```diff
@@ -314,7 +314,7 @@ bool target_direction;
   };
 #endif
 
-#ifdef SERVO_ENDSTOPS
+#if ENABLED(HAS_SERVO_ENDSTOPS)
```
Comment (thinkyhead) [inline Marlin/Marlin_main.cpp:317]: But, if it _was_ set to `true` this line could simply be `#if HAS_SERVO_ENDSTOPS`, which is how all the other `HAS_*` defines work.

**Comment by thinkyhead** [comment]
Indeed, feel free to start using `ENABLED`/`DISABLED` where needed. I will submit more updates to the code around these soon also.

**Comment by Wackerbarth** [comment]
@thinkyhead -- I don't like your use of "defined" as the selector.  I thought that we were moving toward the use of ENABLED and "HAS__" .
However, the #if HAS__\* construct doesn't really work without the ENABLED because a user would want to simply use "#define HAS_SOMETHING" without necessarily adding on the "true"

**Comment by thinkyhead** [comment]
> I don't like your use of "defined" as the selector.

This is a case of doing the least modification just to fix a compile bug. The `HAS_SERVOS` define was just proposed a day or 2 ago.

**Comment by thinkyhead** [comment]
#2554 covers this and cleans up a bit besides…

**Pull request closed without merging**